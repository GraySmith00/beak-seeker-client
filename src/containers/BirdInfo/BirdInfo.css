.bird-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 2rem;
  background-color: #fff;
  height: 1400px; }
  .bird-info .main-content {
    margin: 59px auto; }
  .bird-info .bird-content {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%; }
    .bird-info .bird-content .info {
      width: 90%; }
    .bird-info .bird-content img {
      width: 100%;
      -webkit-box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58);
      -moz-box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58);
      box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58); }
    .bird-info .bird-content h3 {
      margin: 1.5rem 0 0.5rem 0;
      font-size: 1.8rem; }
    .bird-info .bird-content p {
      margin-top: 0; }
