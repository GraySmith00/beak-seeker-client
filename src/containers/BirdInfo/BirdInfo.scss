.bird-info {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 2rem;
  background-color: #fff;

  .main-content {
    margin: 59px auto;
  }

  .bird-content {
    display: flex;
    flex-direction: column;
    align-items: center;

    width: 100%;

    .info {
      width: 90%;
    }

    img {
      width: 100%;
      -webkit-box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58);
      -moz-box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58);
      box-shadow: 0px 11px 38px -10px rgba(0, 0, 0, 0.58);
    }

    h3 {
      margin: 1.5rem 0 0.5rem 0;
      font-size: 1.8rem;
    }

    p {
      margin-top: 0;
    }
  }
}
