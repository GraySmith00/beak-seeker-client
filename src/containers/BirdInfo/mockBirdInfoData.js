export const mockBird = {
  speciesCode: 'cangoo',
  comName: 'Canada Goose',
  sciName: 'Branta canadensis',
  locId: 'L883813',
  locName: 'Rocky Mountain Lake Park',
  obsDt: '2018-09-06 06:40',
  howMany: 15,
  lat: 39.7824006,
  lng: -105.0299138,
  obsValid: true,
  obsReviewed: false,
  locationPrivate: false
};
