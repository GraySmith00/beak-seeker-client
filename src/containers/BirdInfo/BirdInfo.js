import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { getBirdImage, getBirdInfo } from '../../utils/apiCalls';
import LoadingSpinner2 from '../../components/LoadingSpinner2/LoadingSpinner2';
import { connect } from 'react-redux';

import Header from '../../components/Header/Header';

import './BirdInfo.css';

export class BirdInfo extends Component {
  constructor() {
    super();
    this.state = {
      comName: '',
      birdImage: '',
      birdInfo: '',
      loading: true
    };
  }
  async componentDidMount() {
    await this.loadBirdInfo();
  }

  loadBirdInfo = async () => {
    const { locId, speciesCode, hotspots } = this.props;
    const location = hotspots.find(hotspot => hotspot.locId === locId);

    const bird = location.birds.find(bird => bird.speciesCode === speciesCode);
    const { comName } = bird;

    const birdImage = await getBirdImage(comName);
    const birdInfo = await getBirdInfo(comName);

    this.setState({
      comName,
      birdImage,
      birdInfo,
      loading: false
    });
  };

  render() {
    const { birdImage, birdInfo, loading, comName } = this.state;
    let birdContent;

    if (loading) {
      birdContent = <LoadingSpinner2 />;
    } else {
      birdContent = (
        <div className="bird-content">
          {birdImage ? (
            <img src={birdImage} alt="bird" />
          ) : (
            <p className="not-found-text">
              Sorry, we could not find an image of this bird at this time :(
            </p>
          )}
          <div className="info">
            <h3>{comName}</h3>
            <p>{birdInfo}</p>
          </div>
        </div>
      );
    }

    return (
      <div className="bird-info">
        <Header currentPage="Bird Info" />
        <main className="main-content">{birdContent}</main>
      </div>
    );
  }
}

BirdInfo.propTypes = {
  locId: PropTypes.string.isRequired,
  speciesCode: PropTypes.string.isRequired
};

export const mapStateToProps = state => ({
  hotspots: state.hotspots
});

export default connect(mapStateToProps)(BirdInfo);
