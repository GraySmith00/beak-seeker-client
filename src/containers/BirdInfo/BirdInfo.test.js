import React from 'react';
import { shallow } from 'enzyme';
import { BirdInfo } from './BirdInfo';

import { mockBird, mockHotspotsWithBirds } from './mockBirdInfoData';

jest.mock('../../utils/apiCalls');

describe('BirdInfo component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <BirdInfo
        locId={mockBird.locId}
        speciesCode={mockBird.speciesCode}
        hotspots={mockHotspotsWithBirds}
      />,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('should have an initial local state', () => {
    const expected = {
      comName: '',
      birdImage: '',
      birdInfo: '',
      loading: true
    };

    expect(wrapper.state()).toEqual(expected);
  });

  describe('loadBirdInfo', () => {
    it('should set the proper state of the bird that was passed in', async () => {
      const expected = {
        birdImage:
          'https://upload.wikimedia.org/wikipedia/commons/thumb/e/e6/Kanadagans_Branta_canadensis.jpg/200px-Kanadagans_Branta_canadensis.jpg',
        birdInfo:
          'The Canada goose (Branta canadensis) is a large wild goose species with a black head and neck, white cheeks, white under its chin, and a brown body.',
        comName: 'Canada Goose',
        loading: false
      };

      await wrapper.instance().loadBirdInfo();

      expect(wrapper.state()).toEqual(expected);
    });
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
