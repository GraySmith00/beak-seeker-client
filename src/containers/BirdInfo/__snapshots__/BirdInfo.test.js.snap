// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirdInfo component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BirdInfo
    locId="L883813"
    speciesCode="cangoo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Header)
          currentPage="Bird Info"
/>,
        <main
          className="main-content"
>
          <LoadingSpinner />
</main>,
      ],
      "className": "bird-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentPage": "Bird Info",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <LoadingSpinner />,
          "className": "main-content",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Header)
            currentPage="Bird Info"
/>,
          <main
            className="main-content"
>
            <LoadingSpinner />
</main>,
        ],
        "className": "bird-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentPage": "Bird Info",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <LoadingSpinner />,
            "className": "main-content",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
